### PR TITLE
Validate SPV_KHR_workgroup_memory_explicit_layout

### PR DESCRIPTION
**Pull request by cmarcelo**

### CODE REVIEW

File: DEPS [hunk 1]
```diff
@@ -6,7 +6,7 @@ vars = {
   'effcee_revision': '2ec8f8738118cc483b67c04a759fee53496c5659',
   'googletest_revision': '3af06fe1664d30f98de1e78c53a7087e842a2547',
   're2_revision': 'ca11026a032ce2a3de4b3c389ee53d2bdc8794d6',
-  'spirv_headers_revision': 'f027d53ded7e230e008d37c8b47ede7cd308e19d',
+  'spirv_headers_revision': '38fe9b08508c876ba830f250a488c51003531e7e',
```
Comment (cmarcelo) [inline DEPS:9]: NOTE: This will be updated once https://github.com/KhronosGroup/SPIRV-Headers/pull/192 lands.

**Comment by cmarcelo** [comment]
@dneto0 @alan-baker could you take a look at this one?

Note: the third commit is a new addition to allow validating scalar block layout for Workgroup. Since VK allows enabling that independently, it became a separate option.

File: tools/val/val.cpp [hunk 2]
```diff
@@ -58,6 +58,7 @@ NOTE: The validator is a work in progress.
                                    uniform, storage buffer, and push constant layouts.  Scalar layout
                                    rules are more permissive than relaxed block layout so in effect
                                    this will override the --relax-block-layout option.
+  --workgroup-scalar-block-layout  Enable scalar block layout when checking Workgroup block layouts.
```
Comment (alan-baker) [inline tools/val/val.cpp:61]: This needs propagated to spirv-opt as well and spirv-opt will have to pass it along to the validator.

**Review by alan-baker — CHANGES_REQUESTED** [review]
The optimizer runs validation before any passes by default so the new option needs to be exposed that way too.

File: tools/val/val.cpp (hunk 2 above)
Comment (cmarcelo) [reply tools/val/val.cpp:61]: Done.

**Comment by cmarcelo** [comment]
@alan-baker windows failures seem about CI environment not having cmake around (?), an excerpt from one of the logs:

```
T:\src\github\SPIRV-Tools>if 2013 == 2017 (
call "C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Auxiliary\Build\vcvarsall.bat" x64
 echo "Using VS 2017..."
)  else if 2013 == 2015 (
call "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\vcvarsall.bat" x64
 echo "Using VS 2015..."
)  else if 2013 == 2013 (
call "C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\vcvarsall.bat" x64
 echo "Using VS 2013..."
)
"Using VS 2013..."
"Starting build... Wed 01/27/2021  9:31:53.59"
'cmake' is not recognized as an internal or external command,
operable program or batch file.


[ID: 4347538] Build finished after 16 secs, exit value: 49


Warning: Permanently added 'localhost' (ECDSA) to the list of known hosts.
[09:31:54] Collecting build artifacts from build VM
Build script failed with exit code: 49
```

**Comment by alan-baker** [comment]
Yeah I'm trying to figure out what's gone wrong.

**Comment by alan-baker** [comment]
I've fixed the bot issue, please rebase off master to pick up the change.

**Pull request merged**